### ISSUE TITLE
Proposal: remove implicit `callconv(.C)` on `extern` functions

### ISSUE THREAD

**Issue by tadeokondrak** [ziglang/zig#5269]
Currently, these do the same thing:

0. `extern fn foo() void;`
1. `extern fn foo() callconv(.C) void;`

Callconv didn't exist before, so extern was the only way to mark the calling convention. Now that we have callconv, the implicit calling convention isn't needed anymore.

**Comment by emekoi** [comment]
> Communicate intent precisely.
> Only one obvious way to do things.

**Comment by pixelherodev** [comment]
This would require a bit more work, but overall, I'm in favor of this.

Calling convention _must_ then be defined. The default calling convention is "unspecified." For obvious reasons, this isn't usable for exported / extern functions (note: this proposal must also apply to `export` for the same reason). What should be needed instead is a valid, defined calling convention, which can be used both for exports and imports without issues.

**Comment by tadeokondrak** [comment]
I see the accepted label has been removed, and I don't support this change anymore.

Learning from the `inline`/`callconv(.Inline)` change and its reversal, I think that it sometimes makes sense for the syntax used the vast majority of the time to be a special case, and have a secondary syntax when it makes sense.

In the case of `inline` and `extern`, they're much more likely to be used than any other `callconv` expression, so it makes sense to use those, but keep the callconv syntax for other calling conventions that don't deserve their own keyword, and for selecting the callconv of a function based on a comptime variable.

I realized that this could apply to #1717 too, where the length of the syntax
